restapi/endpoints_test.go

```diff
@@ -0,0 +1,197 @@
+package restapi
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gavv/httpexpect/v2"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/suite"
+	"github.com/uwblueprint/shoe-project/internal/database/migrations"
+	"github.com/uwblueprint/shoe-project/internal/database/models"
+	"github.com/uwblueprint/shoe-project/testutils"
+	"gorm.io/gorm"
+)
+
+type endpointTestSuite struct {
+	suite.Suite
+	endpoint *httpexpect.Expect
+	db       *gorm.DB
+	token    string
+}
+
+func (suite *endpointTestSuite) SetupSuite() {
+	db, err := testutils.MockDatabase()
+	if err != nil {
+		suite.Fail("error while creating database", err)
+	}
+	suite.db = db
+
+	// setup required for jwt authentication
+	viper.SetDefault("auth.jwt_key", "random")
+	viper.SetDefault("auth.jwt_expiry", "5m")
+	viper.SetDefault("auth.jwt_issuer", "endpoint_tests")
+
+	router, err := Router(db)
+	if err != nil {
+		suite.FailNow("error while creating router", err)
+	}
+
+	server := httptest.NewServer(router)
+	suite.endpoint = httpexpect.New(suite.T(), server.URL)
+}
+
+func (suite *endpointTestSuite) SetupTest() {
+	if err := migrations.CreateTables(suite.db); err != nil {
+		suite.Fail("error while creating tables", err)
+	}
+	if err := migrations.CreateSuperUser(suite.db); err != nil {
+		suite.Fail("error creating super user", err)
+	}
+
+	suite.token = suite.endpoint.POST("/login").
+		WithJSON([]models.User{{
+			Username: "admin",
+			Password: "root",
+		}}).
+		Expect().
+		Status(http.StatusOK).JSON().Object().Value("payload").String().Raw()
+}
+
+func (suite *endpointTestSuite) TearDownTest() {
+	if err := testutils.DropTables(suite.db); err != nil {
+		suite.Fail("error while dropping tables", err)
+	}
+}
+
+func (suite *endpointTestSuite) TestHealthCheck() {
+	suite.endpoint.GET("/health").
+		Expect().
+		Status(http.StatusOK).JSON().Object().Value("message").String().Equal("Hello World")
+}
+
+func (suite *endpointTestSuite) TestGetAllStories() {
+	json_story1 := []models.Story{
+		{
+			Title:    "The Little Prince",
+			Content:  "Children",
+			AuthorID: 1,
+		},
+	}
+
+	json_story2 := []models.Story{
+		{
+			Title:    "Hitchhiker's Guide to the Galaxy",
+			Content:  "Fiction",
+			AuthorID: 2,
+		},
+	}
+
+	//Directly add to mock DB
```

Comment:
side note: we're not mocking the db anymore, more like a sleek/light db (using sqlite)

so really small nit: we should remove references to "mock db" like in this comment and the `mock` var name below (unless thats convention, im not really sure)